staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -163,16 +163,16 @@ func (t *prefixTransformers) TransformFromStorage(ctx context.Context, data []by
 		logTransformErr(ctx, err, "failed to decrypt data")
 		return nil, false, err
 	}
-	RecordTransformation("from_storage", "unknown", time.Since(start), t.err)
+	RecordTransformation(resource, "from_storage", "unknown", time.Since(start), t.err)
 	return nil, false, t.err
 }
 
 // TransformToStorage uses the first transformer and adds its prefix to the data.
-func (t *prefixTransformers) TransformToStorage(ctx context.Context, data []byte, dataCtx Context) ([]byte, error) {
+func (t *prefixTransformers) TransformToStorage(ctx context.Context, resource string, data []byte, dataCtx Context) ([]byte, error) {
```

Comment:
Undo changes to the transformer interface and use `genericapirequest.RequestInfoFrom` to fetch the resource.

Reply:
i don't see the requestinfo present in context for a lot of requests

Reply:
We aren't going to expand the transformer interface, so you will need to sort out the wiring to plumb that via the context

Reply:
any specific reason for not expanding the interface?

Reply:
> any specific reason for not expanding the interface?

Because no transformer does anything related to transforming bytes based on the that input.  It has nothing to do with the interface.